src/v2/params/types.ts

```diff
@@ -69,31 +72,36 @@ export interface SelectOptions<T = unknown> {
   value: T;
 }
 
-export interface ParamSpec<T = unknown> {
+export type ParamSpec<T = unknown> = {
   name: string;
   default?: T;
   label?: string;
-  as?: string;
   description?: string;
   type: ParamValueType;
-  input?: ParamInput<T>;
-}
+} & HasAtMostOneInput<T>;
 
 export type ParamOptions<T = unknown> = Omit<ParamSpec<T>, 'name' | 'type'>;
 
-export class Param<T = unknown> {
+export abstract class Param<T extends string | number | boolean | string[]> {
   static type: ParamValueType = 'string';
 
   constructor(readonly name: string, readonly options: ParamOptions<T> = {}) {}
 
-  get rawValue(): string | undefined {
-    return process.env[this.name];
+  get value(): T {
+    throw new Error('Not implemented');
+  }
+
+  expr() {
+    return new ParamExpression<T>(this);
```

Comment:
Can you explain what punchline you're setting up here with ParamExpression<T>?

I originally intended expr to just be a string representation, so I want to make sure I'm following what's going on here.

Reply:
Nothing good, that's what. Got rid of ParamExpression